deeppavlov/core/trainers/fit_trainer.py

```diff
@@ -0,0 +1,260 @@
+
+log = get_logger(__name__)
+
+
+@register('fit_trainer')
+class FitTrainer:
+    """
+    Trainer class for fitting and evaluating :class:`Estimators <deeppavlov.core.models.estimator.Estimator>`
+
+    Args:
+        chainer_config: ``"chainer"`` block of a configuration file
+        batch_size: batch_size to use for partial fitting (if available) and evaluation,
+            the whole dataset is used if ``batch_size`` is negative or zero (default is ``-1``)
+        metrics: iterable of registered metrics names with optional lists of their inputs from chainer config
+            (default is ``('accuracy',)``)
+        evaluation_targets: data types on which to evaluate trained pipeline (default is ``('valid', 'test')``)
+        show_examples: a flag used to print inputs, expected outputs and predicted outputs for the last batch
+            in evaluation logs (default is ``False``)
+        tensorboard_log_dir: path to a directory where tensorboard logs can be stored, ignored if None
+            (default is ``None``)
+        max_test_batches: maximum batches count for pipeline testing and evaluation, ignored if negative
+            (default is ``-1``)
+        **kwargs: additional parameters whose names will be logged but otherwise ignored
+    """
+    def __init__(self, chainer_config: dict, *, batch_size: int = -1,
+                 metrics: Iterable[Union[str, dict]] = ('accuracy',),
+                 evaluation_targets: Iterable[str] = ('valid', 'test'),
+                 show_examples: bool = False,
+                 tensorboard_log_dir: Optional[Union[str, Path]] = None,
+                 max_test_batches: int = -1,
+                 **kwargs) -> None:
+        if kwargs:
+            log.info(f'{self.__class__.__name__} got additional init parameters {list(kwargs)} that will be ignored:')
+        self.chainer_config = chainer_config
+        self._chainer = Chainer(chainer_config['in'], chainer_config['out'], chainer_config.get('in_y'))
+        self.batch_size = batch_size
+        self.metrics = parse_metrics(metrics, self._chainer.in_y, self._chainer.out_params)
+        self.evaluation_targets = tuple(evaluation_targets)
+        self.show_examples = show_examples
+
+        self.max_test_batches = None if max_test_batches < 0 else max_test_batches
+
+        self.tensorboard_log_dir: Optional[Path] = tensorboard_log_dir
+        if tensorboard_log_dir is not None:
+            try:
+                # noinspection PyPackageRequirements
+                # noinspection PyUnresolvedReferences
+                import tensorflow
+            except ImportError:
+                log.warning('TensorFlow could not be imported, so tensorboard log directory'
+                            f'`{self.tensorboard_log_dir}` will be ignored')
+                self.tensorboard_log_dir = None
+            else:
+                self.tensorboard_log_dir = expand_path(tensorboard_log_dir)
+                self._tf = tensorflow
```

Comment:
Why not `import tensorflow as tf`?

Reply:
It will be a pain to manage between different trainers. Can be resolved by using `tensorboardX`, but in another pull-request